internal/backend/remote-state/s3/client.go

```diff
@@ -226,17 +228,31 @@ func (c *RemoteClient) Lock(info *statemgr.LockInfo) (string, error) {
 
 		info.ID = lockID
 	}
+	item := map[string]*dynamodb.AttributeValue{
+		"LockID": {S: aws.String(c.lockPath())},
+		"Info":   {S: aws.String(string(info.Marshal()))},
+	}
 
 	putParams := &dynamodb.PutItemInput{
-		Item: map[string]*dynamodb.AttributeValue{
-			"LockID": {S: aws.String(c.lockPath())},
-			"Info":   {S: aws.String(string(info.Marshal()))},
-		},
+		Item:                item,
 		TableName:           aws.String(c.ddbTable),
 		ConditionExpression: aws.String("attribute_not_exists(LockID)"),
 	}
-	_, err := c.dynClient.PutItem(putParams)
 
+	if c.ddbLockTTL != 0 {
+		ttlAttributeName := "ttl"
+		ttl := time.Now().Unix() + int64(c.ddbLockTTL)
+		item[ttlAttributeName] = &dynamodb.AttributeValue{N: aws.String(strconv.FormatInt(ttl, 10))}
```

Comment:
Is this code the implementation or the test? I see a hardcoded `10` value here. Is that supposed to be using the configuration value or is that expected to be 10? I noticed the test confirms 10 second TTL was used but not sure if this should be here or not.

Reply:
Looks like the second argument is the number's base, from https://pkg.go.dev/strconv#FormatInt